src/NuGet.Services.BasicSearch/Startup.cs

```diff
@@ -91,6 +91,26 @@ public class Startup
 
             _logger.LogInformation(LogMessages.AppStartup);
 
+            // Overwrite the index's Azure Directory cache path if configured ot use an Azure Local Storage resource.
+            if (!string.IsNullOrEmpty(config.AzureDirectoryCacheLocalResourceName))
+            {
+                if (!SafeRoleEnvironment.IsAvailable)
+                {
+                    _logger.LogWarning(
+                        "Cannot use Azure Local Resource {LocalResourceName} for caching when the RoleEnvironment is not available",
+                        config.AzureDirectoryCacheLocalResourceName);
+                }
+                else
+                {
+                    config.AzureDirectoryCachePath = SafeRoleEnvironment.GetLocalResourceRootPath(config.AzureDirectoryCacheLocalResourceName);
```

Comment:
Let's document on `AzureDirectoryCachePath` that it gets overwritten at runtime if `AzureDirectoryCacheLocalResourceName` is set.

Reply:
Fixed